server/commands/connected_objects.go

```diff
@@ -69,26 +69,31 @@ func (c *ConnectedObjectsCommand) streamedConnectedObjects(
 	}
 
 	for _, ingress := range ingresses {
```

Comment:
uhhhh i'm a bit nervous about changing this, because i would have to make sure that this version doesn't introduce more leaks 😅   i would prefer to keep it as is

Reply:
I'm nervous with these changes as is because of the serial context cancellation. It also seems like magic and isn't obvious to the developer how/why the change fixes the issue.

I think the minor refactor is well worth the investment in the time spent now because it's fresh in our minds.

Reply:
Ok, ran a quick load test with this change and it seems to work fine.

Reply:
Let's do this refactor shortly after merging this. At least this will fix an issue in the meantime and then we can work through a more clear refactor right after this.